### PR TITLE
More Cafe stuff

### PR DESCRIPTION
**Pull request by NigglyWiggly**
Only one actual cafe drink, but a few more lively variations of the
gigapuddi, two ice cream desserts that are made in a microwave because
spess, a quick edit to the booze dispenser, and corndogs!

### CODE REVIEW

File: code/datums/recipe.dm [hunk 1]
```diff
@@ -96,6 +96,7 @@
 //Note : Due to changes to no longer wipe nutriments from cooked items, this is the same as make. So from now on this is THE "turn recipe into new thing" proc
 /datum/recipe/proc/make_food(var/obj/container as obj) //Find our recipe machine and let's begin
 	var/obj/result_obj = new result(container) //Spawn the result of our little cuisine in the recipe machine in advance to transfer reagents
+	world << "[result_obj] \ref[result_obj]"
```
Comment (ComicIronic) [inline code/datums/recipe.dm]: Remove this

File: code/modules/food/recipes_microwave.dm [hunk 2]
```diff
@@ -1400,6 +1400,32 @@
 	reagents = list("milk" = 15, "sodiumchloride" = 5)
 	result = /obj/item/weapon/reagent_containers/food/snacks/gigapuddi/anger
 
+//LIVING PUDDI
+//This is a terrible idea.
+
+/datum/recipe/livingpuddi
```
Comment (ComicIronic) [inline code/modules/food/recipes_microwave.dm]: This has to go at the end of this section.
Due to the way the recipes are generated and checked in the code, the first good recipe (this one) will be selected for all the ingredient combinations, so the sub-recipes won't complete.

File: code/modules/food/recipes_microwave.dm [hunk 3]
```diff
@@ -1470,6 +1500,38 @@
...
+		/obj/item/weapon/reagent_containers/food/snacks/egg,
+		/obj/item/weapon/reagent_containers/food/snacks/sausage,
+		/obj/item/weapon/reagent_containers/food/snacks/grown/corn)
+	result = /obj/item/weapon/reagent_containers/food/snacks/corndog
+
+/datum/recipe/cornydog
+	items = list(
+		/obj/item/weapon/reagent_containers/food/snacks/grown/corn,
+		/obj/item/weapon/reagent_containers/food/snacks/meat/corgi,
+		/obj/item/stack/rods)
```
Comment (ComicIronic) [inline code/modules/food/recipes_microwave.dm]: Make this a kebab thingy instead.

File: code/modules/mob/living/simple_animal/puddi.dm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	icon_state = "livingpuddi"
+	icon_living = "livingpuddi"
+	icon_dead = "livingpuddi-dead"
+	health = 30
+	maxHealth = 30
+	speak = list("Puddi!")
+	speak_emote = list("shouts")
+	emote_hear = list("shouts")
+	emote_see = list("shouts")
+	speak_chance = 1
```
Comment (ComicIronic) [inline code/modules/mob/living/simple_animal/puddi.dm]: Unless it has something to say, this should really be 0.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 5]
```diff
@@ -3350,6 +3353,25 @@
...
+	New()
+		..()
+		reagents.add_reagent("nutriment", 5)
+		bitesize = 5
+
+/obj/item/weapon/reagent_containers/food/snacks/cornydog
+	name = "CORNY DOG"
+	desc = "This is just ridiculous."
+	icon_state = "cornydog"
+	trash = /obj/item/weapon/arrow/rod
```
Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: This should be the thing you put in for the recipe.

File: code/modules/food/recipes_microwave.dm (hunk 2 above)
Comment (NigglyWiggly) [inline code/modules/food/recipes_microwave.dm]: Really? I had no trouble with it when I tried to make one the last time I tested it, but I'll change it regardless

**Pull request merged**